[RFC] Surround pagination with nav element

cliffparnitzky opened the pull request:
Closes #775

leofeyer commented on src/Resources/contao/templates/backend/be_pagination.html5:
```diff
@@ -1,22 +1,24 @@
-      <li class="last"><a href="<?= $this->last['href'] ?>" class="last" title="<?= $this->last['title'] ?>"><?= $this->last['link'] ?></a></li>
-    <?php endif; ?>
-  </ul>
+  <nav role="navigation" aria-label="<?= $GLOBALS['TL_LANG']['MSC']['paginationAriaLabel'] ?>">
```
I would prefer to replace the surrounding `<div>` with the `<nav>` element instead of adding another nesting level.

leofeyer commented on src/Resources/contao/languages/en/default.xlf:
```diff
@@ -1679,6 +1679,9 @@ This e-mail has been generated by Contao. You can not reply to it directly.
         <source>You have to add a root page in the site structure to store file meta information.</source>
       </trans-unit>
+      <trans-unit id="MSC.paginationAriaLabel">
+        <source>Pagination</source>
```
Again, what is the added value here?

NinaG replied:
If there are several navigation elements on one page, it is very helpful for screen reader users if you label them differently so that the user is able to distinguish their usage. Something like "main navigation", "breadcrumb navigation" etc.

leofeyer replied:
@cliffparnitzky Can you change the label to "Pagination menu" please?

cliffparnitzky commented:
Ok, thanks. To finalize the discussion, here my summary of changes:

- remove german translation
- replace the surrounding `<div>` with the `<nav>` element (add the `class` attribute of the `div`)

Correct?

leofeyer commented:
I guess we should also add a [skip link](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/templates/backend/be_main.html5#L57) while we are on it.

cliffparnitzky commented:
I have replaced the surrounding div with the nav element.

I also added the skip link. The probleme here is the generic label. I don't know if it is the best solution using the `$GLOBALS` in the template. If not, the part in `DC_Table` and `Versions` has to be modified.

leofeyer commented on src/Resources/contao/templates/backend/be_pagination.html5:
```diff
@@ -1,8 +1,9 @@
 
   <p><?= $this->total ?></p>
-
+    
```
I think you have to revert this change.

cliffparnitzky replied:
@leofeyer whats exactly? The empty row, the blanks or the complete template change?

leofeyer replied:
The line indentation.

cliffparnitzky replied:
Okay, removed it.

leofeyer commented on src/Resources/contao/templates/backend/be_pagination.html5:
```diff
@@ -1,5 +1,6 @@
 
-<div class="pagination <?= $this->class ?>">
+<nav class="pagination <?= $this->class ?>" role="navigation" aria-label="<?= $GLOBALS['TL_LANG']['MSC']['paginationAriaLabel'] ?>">
+  <a href="<?= $this->request ?>#skipNavigation" class="invisible"><?= \StringUtil::specialchars($GLOBALS['TL_LANG']['MSC']['skipNavigation']); ?></a>
```
This is not correct. First of all, you must not use a hard coded skip ID, because there might be more than one pagination menu and it would lead to ambiguous IDs. Second of all, you also have to add the target element: 

https://github.com/contao/core-bundle/blob/master/src/Resources/contao/templates/modules/mod_navigation.html5#L13

cliffparnitzky replied:
@leofeyer Am i right, or is anything to do here?

leofeyer replied:
> or is anything to do here?

Of course there is:

> First of all, you must not use a hard coded skip ID, because there might be more than one pagination menu and it would lead to ambiguous IDs. Second of all, you also have to add the target element:

None of this has been fixed, has it?

cliffparnitzky replied:
@leofeyer What about my comment?

> We are talking about the backend here, aren't we?

leofeyer replied:
There might be more than one pagination menu in the back end, too.

cliffparnitzky replied:
@leofeyer Now, I've added the target element for the naviation skipping.

BUT: I'm still of the opinion, that multiple paginations in the back end are not possible. In the current implementation there is no unique identifier for a pagination. And scrolling will be done via URL parameter `lp` ... if there is more than one pagination no decision can be done, which list to scroll.

cliffparnitzky commented on src/Resources/contao/templates/backend/be_pagination.html5:
```diff
@@ -1,5 +1,6 @@
 
-<div class="pagination <?= $this->class ?>">
+<nav class="pagination <?= $this->class ?>" role="navigation" aria-label="<?= $GLOBALS['TL_LANG']['MSC']['paginationAriaLabel'] ?>">
+  <a href="<?= $this->request ?>#skipNavigation" class="invisible"><?= \StringUtil::specialchars($GLOBALS['TL_LANG']['MSC']['skipNavigation']); ?></a>
```
At first sorry for my late reply.

To your statement above: I think thats the same like her: https://github.com/contao/core-bundle/pull/868#issuecomment-307322372

We are talking about the backend here, aren't we?

leofeyer commented:
Implemented in 432ca9d77cbf0b1d2dcc8035a765cec6df172331.
